sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -875,9 +859,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              *
              * However, we support a single column RowType through the SingleColumnTableType.
              */
-            Field field = fields.get(0);
-            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            return new SelectSymbol(relation, singleColumnTableType);
+            DataType innerType = fields.get(0).valueType();
+            SingleColumnTableType dataType = new SingleColumnTableType(innerType);
+            final SelectSymbol.ResultType resultType;
+            if (context.isArrayComparisonChild(node)) {
+                resultType = SelectSymbol.ResultType.SINGLE_COLUMN_MULTIPLE_VALUES;
```

Comment:
This would duplicate code (creation of SelectSymbol) in order to save one line of code. Nothing terrible wrong with that but IMHO a purely aesthetic code style matter 🤓